Accept FileList as Rake task source files

avdgaag opened the pull request:
The Rake task currently insists on taking a single glob pattern, while Rake's FileList is perfectly capable of taking multiple patterns. By relaxing the type check and taking advantage of the fact that passing a `FileList` to `FileList.[]` returns the same `FileList`, we can allow developers to set a fully customized `FileList` without changing the original API.

The right place to test this is unclear to me. There are no unit tests for the Rake task, and only some integration tests. Any suggestions?

See #788.

troessner commented on lib/reek/rake/task.rb:
```diff
@@ -85,7 +85,9 @@ def initialize(name = :reek)
       def source_files=(files)
-        raise ArgumentError, no_string_given_for_file_list_warning unless files.is_a?(String)
+        unless files.is_a?(String) || files.is_a?(FileList)
+          raise ArgumentError, no_string_or_filelist_given_for_file_list_warning
```
Let's rather drop this ugly method (I know it was there before, but still) and just use:

```Ruby
raise ArgumentError, "File list should be a FileList or a String that can contain"\
  "a glob pattern, e.g. '{app,lib,spec}/**/*.rb'"
```

troessner commented:
>> There are no unit tests for the Rake task, and only some integration tests. 

An integration test would be good enough here.
Can you also please update our [Rake task docs](https://github.com/troessner/reek/blob/master/docs/Rake-Task.md)?

avdgaag commented:
@troessner I added a test and some docs. What do you think?

troessner commented on docs/Rake-Task.md:
```diff
@@ -56,3 +64,5 @@ REEK_CFG="config/custom.reek" REEK_OPTS="-s" rake reek
 
 See also: [Reek-Driven-Development](Reek-Driven-Development.md)
+
+[Rake::FileList]: http://rake.rubyforge.org/classes/Rake/FileList.html
```
Why not use GFM and then do [link text](link)?

troessner commented on docs/Rake-Task.md:
````diff
@@ -41,6 +41,14 @@ Reek::Rake::Task.new do |t|
 end
 ```
 
+Alternatively, you can create your own [Rake::FileList][] and use that for `source_files`:
````
Lets rather use GFM for links, see below.

troessner commented:
Nice job!
Please update with the 2 tiny changes mentioned above and squash to one commit, then we can merge this one!

avdgaag commented:
I’m note sure what you mean by a “GFM link” and a Markdown link — as far as I know, they’re the same. But I’ve changed my link to use the inline-style, guessing that is what you are referring to. I’ve also squashed the commits together.

troessner commented:
Yes, that's what i meant! ;)
Merging, thanks! 
Release upcoming in 20 minutes.

avdgaag commented:
Great, thanks!

troessner commented:
Aaaand 3.8.3 is released including this change ;)
